### PR TITLE
Add `managed_identity_id` to `databricks_storage_credential` to support user-assigned managed identities

### LINKED ISSUE #2532: [FEATURE] Add support of User Managed Identity (UMI) to azure's databricks_storage_credential
### Use-cases
On Azure I would like to add storage credentials which are backed by **User Managed Identity**, currently  `databricks_storage_credential` supports only MI and SPN, but there is no support for UMI.

### Proposal
Judging by API docs at https://docs.databricks.com/api/azure/workspace/storagecredentials/create there is extra parameter that needs to be passed to make it working: `azure_managed_identity.managed_identity_id`

Current TF resource declaration looks like this:
```
resource "databricks_storage_credential" "external_mi" {
  name = "mi_credential"
  azure_managed_identity {
    access_connector_id = "the string pointing to access connector resource id"
  }
  comment = "Managed identity credential managed by TF"
}
```

With UMI would look like this:
```
resource "databricks_storage_credential" "external_mi" {
  name = "mi_credential"
  azure_managed_identity {
    access_connector_id = "the string pointing to access connector resource id"
    managed_identity_id = "the string pointing to user defined managed identity id"
  }
  comment = "Managed identity credential managed by TF"
}
```
just the extra `managed_identity_id` would have to be passed though to the API.[ GO SDK supports it already](https://github.com/databricks/databricks-sdk-go/blob/6e78f494c993099a6f5f5209caa203df5b1e2fef/service/catalog/model.go#L82)

### References
- https://docs.databricks.com/api/azure/workspace/storagecredentials/create
- https://github.com/databricks/databricks-sdk-go/blob/6e78f494c993099a6f5f5209caa203df5b1e2fef/service/catalog/model.go#L82

### PR DESCRIPTION
**Pull request by alexott**
## Changes

This fixes #2532 

@nkvuong I think that the metastore data access should be also updated, but I couldn't find documentation on that API

## Tests

- [x] `make test` run locally
- [x] relevant change in `docs/` folder
- [ ] covered with integration tests in `internal/acceptance`
- [x] relevant acceptance tests are passing
- [x] using Go SDK

### CODE REVIEW

**Comment by nkvuong** [comment]
@alexott can we hold off this one until #2182 is merged, as that PR is refactoring storage credential to use the SDK

**Comment by alexott** [comment]
ah, yes.

File: catalog/resource_metastore_data_access.go
Comment (mgyucht) [inline catalog/resource_metastore_data_access.go:32]: Can we use Go SDK?

**Comment by alexott** [comment]
Marking as draft until #2182 is merged

File: catalog/resource_metastore_data_access.go
Comment (tanmay-db) [inline catalog/resource_metastore_data_access.go:63]: I think if it's used by both resources then we should probably move it out to other common file for better readability.

**Review by tanmay-db — APPROVED** [review]
LGTM

**Comment by tanmay-db** [comment]
Hi @kuntaldeb-upl, we are going to do a release shortly and will update you once it is done

**Comment by tanmay-db** [comment]
Hi @kuntaldeb-upl, yes it will be available in v1.25.0 release.

**Pull request merged**
